### ISSUE TITLE
Parameterize Release Manager timeout

### ISSUE THREAD

**Issue by felipecruz91** [opendevstack/ods-jenkins-shared-library#452]
**Is your feature request related to a problem? Please describe.**
Some containers may take a long time to start-up and change from `Waiting (Container creating)` to `Running` state.

A couple of scenarios when this could happen are:

- When the pull image operation may take a long time due to a large image size or bad internet connection.
- When the CPU assigned to the container (and/or its limit) is low and therefore there are not enough CPU cycles for the container to start up within the 60 second window.
- The cluster is at its max capacity and it takes longer to schedule the container into a node with enough resources.

I have noticed that, when using Release Manager to deploy an app, there is a [fixed timeout](https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/services/OpenShiftService.groovy#L285) to wait for the container to progress into a "Running" state. This fixed timeout is set to 5 retries * 12 seconds = 60 seconds.

**Describe the solution you'd like**
A timeout of 60 seconds may seem good enough, however, being pragmatic I have already seen the case where some containers actually take longer than 60 seconds to be in a `Running` state and that's something that we cannot control unfortunately. I'd suggest to keep those values as the default option but also give the option to override them. Ideally, we could have 2 env. vars:

- `MAX_RETRIES`
- `MAX_SECONDS_PER_RETRY`

Let me know your thoughts 😀Thanks!

**Comment by michaelsauter** [comment]
Sounds like a good idea to parameterise. The only issue I see is that this would likely need to be a per-project setting, as I don't see a way to specify this per-component within the current design. But better than nothing. Also, it could be implemented with 60 seconds as the default if the env vars are not set, just as you suggest.

**Comment by tbugfinder** [comment]
Hi,
IMHO the title should be updated slightly as the issue isn't only related to release manager but build/deploy stages.

I'm adding Jenkins build log so that other users can find this issue in case their pipeline also failed related to this issue.
```
.....
[Pipeline] echo
2020-10-01 02:49:35Z Could not find 'running' pod for deployment=xxxxxxxxxxxxxxxxxx - waiting
[Pipeline] sleep
Sleeping for 12 sec
[Pipeline] }
[Pipeline] // stage
.........
.........
[Pipeline] End of Pipeline
java.lang.RuntimeException: Could not find 'running' pod for deployment=myproject-32
	at org.ods.services.OpenShiftService.getPodDataForDeployment(OpenShiftService.groovy:445)
	at org.ods.component.RolloutOpenShiftDeploymentStage.run(RolloutOpenShiftDeploymentStage.groovy:117)
	at org.ods.component.Stage.execute(Stage.groovy:23)
```

**Comment by tbugfinder** [comment]
Shouldn't this be part of 3.1 ?

**Comment by clemensutschig** [comment]
Dupe of #465 and fixed in #467 amd #468
